dts/arm/ambiq/ambiq_apollo4p.dtsi

```diff
@@ -90,6 +91,42 @@
 		pinctrl: pin-controller@40010000 {
 			compatible = "ambiq,apollo4-pinctrl";
 			reg = <0x40010000 0x800>;
+
+			gpioa: gpioa {
+				compatible = "ambiq,gpio";
+				gpio-controller;
+				#gpio-cells = <2>;
+				pin-offset = <0>;
+				interrupts = <56 0>;
+				status = "disabled";
+			};
+
+			gpiob: gpiob {
+				compatible = "ambiq,gpio";
+				gpio-controller;
+				#gpio-cells = <2>;
+				pin-offset = <32>;
```

Comment:
is pin-offset representing an offset from the pinctrl reg entry? if so that looks wrong @mbolivar-ampere @galak